src/machine/machine-dbus.c

```diff
@@ -182,154 +182,30 @@ int bus_machine_method_get_addresses(sd_bus_message *message, void *userdata, sd
         if (r < 0)
                 return r;
 
-        switch (m->class) {
-
-        case MACHINE_HOST: {
-                _cleanup_free_ struct local_address *addresses = NULL;
-                int n;
-
-                n = local_addresses(NULL, 0, AF_UNSPEC, &addresses);
-                if (n < 0)
-                        return n;
-
-                for (int i = 0; i < n; i++) {
-                        r = sd_bus_message_open_container(reply, 'r', "iay");
-                        if (r < 0)
-                                return r;
-
-                        r = sd_bus_message_append(reply, "i", addresses[i].family);
-                        if (r < 0)
-                                return r;
-
-                        r = sd_bus_message_append_array(reply, 'y', &addresses[i].address, FAMILY_ADDRESS_SIZE(addresses[i].family));
-                        if (r < 0)
-                                return r;
-
-                        r = sd_bus_message_close_container(reply);
-                        if (r < 0)
-                                return r;
-                }
-
-                break;
-        }
-
-        case MACHINE_CONTAINER: {
-                _cleanup_close_pair_ int pair[2] = EBADF_PAIR;
-                _cleanup_close_ int netns_fd = -EBADF;
-                pid_t child;
+        r = machine_get_addresses(m, &addresses);
+        if (r == -ENONET)
+                return sd_bus_error_setf(error, BUS_ERROR_NO_PRIVATE_NETWORKING, "Machine %s does not use private networking", m->name);
+        if (r == -EOPNOTSUPP)
+                return sd_bus_error_set(error, SD_BUS_ERROR_NOT_SUPPORTED, "Requesting IP address data is only supported on container machines.");
+        if (r < 0)
+                return sd_bus_error_set_errno(error, r);
 
-                r = in_same_namespace(0, m->leader.pid, NAMESPACE_NET);
+        for (int i = 0; i < r; i++) {
+                r = sd_bus_message_open_container(reply, 'r', "iay");
```

Comment:
This seems broken actually, you're using `r` in the loop condition but then reassigning it in the for loop body, please save `r` in a separate variable before starting the loop and use that in the loop condition.